app/src/main/java/study_dev/testbed/studyhopper/GroupAdapter.java

```diff
@@ -0,0 +1,84 @@
+package study_dev.testbed.studyhopper;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.firebase.ui.firestore.FirestoreRecyclerAdapter;
+import com.firebase.ui.firestore.FirestoreRecyclerOptions;
+
+public class GroupAdapter extends FirestoreRecyclerAdapter<Group, GroupAdapter.GroupHolder> {
+
+    public GroupAdapter(@NonNull FirestoreRecyclerOptions<Group> options) {
+        super(options);
+    }
+
+    @Override
+    protected void onBindViewHolder(@NonNull GroupHolder holder, int position, @NonNull Group model) {
+        holder.textViewGroupName.setText(model.getGroupName());
+        holder.textViewCourseCode.setText(model.getCourseCode());
+        holder.imageViewGroupColor.setImageResource(findGroupColorId(model.getGroupColor()));
+    }
+
+    @NonNull
+    @Override
+    public GroupHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+
+        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.group_card,
+                parent, false);
+
+        return new GroupHolder(v);
+    }
+
+    class GroupHolder extends RecyclerView.ViewHolder {
+        TextView textViewGroupName;
+        TextView textViewCourseCode;
+        ImageView imageViewGroupColor;
+
+        public GroupHolder(@NonNull View itemView) {
+            super(itemView);
+            textViewGroupName = itemView.findViewById(R.id.text_view_group_name);
+            textViewCourseCode = itemView.findViewById(R.id.text_view_course_code);
+            imageViewGroupColor = itemView.findViewById(R.id.image_view_group_color);
+        }
+
+    }
+
+    private int findGroupColorId(String color){
+        if(color.equals("Red")){
```

Comment:
maybe a map would be better? instead of doing a bunch of if statements, pass in color as key and return value. But this is fine as is.

Reply:
Adjusted to utilize map. Old code is commented out for now.